packages/babel/src/transformation/file/options/config.json

```diff
@@ -240,6 +236,12 @@
     "description": "set `sources[0]` on returned source map"
   },
 
+  "inputSourceMap": {
+    "type": "boolean",
+    "description": "use existing source map for source map generation - only works with inline source maps",
```

Comment:
You shouldn't be using this option via the CLI. This option expects an object containing an input source map which you can't really pass over the CLI. If your input code has an inline source map comment at the bottom then this option is populated with the deserialised version of it automatically so the source map inheritance should be working fine. If it's reporting incorrect line information still then it's likely that #2336 is the cause of that.